### PR TITLE
feat[w2.2]: encrypt jmbg in url

### PR DESCRIPTION
**Pull request by draganacarapic**
* I added two new methods for jmbg encryption and decryption, due to the protection of personal data and the impossibility of viewing the patient's jmbg in the url path of the mail activation page,
* Also in the second commit I amended the PatientControllerTests, in order to adhere to the principles of the DRY, in accordance with the professor's suggestion.

### CODE REVIEW

File: HealthcareSystem/PatientWebApp/wwwroot/js/activate_account.js [hunk 1]
```diff
@@ -46,4 +48,34 @@
     });
     });
     
-});
\ No newline at end of file
+});
+
+function encryptJmbg(jmbg) {
```
Comment (milana-todorovic) [inline HealthcareSystem/PatientWebApp/wwwroot/js/activate_account.js]: Wouldn't this be decryptJmbg? Also, I'm not sure putting the decryption algorithm in javascript is a good idea, maybe sending it still encrypted to the activate endpoint and decrypting it there would be better? I'm definitely not an expert on security though. 
Also, if you do end up also moving decryption to the backend, maybe consider making a small EncryptionService to take over this responsibility.

**Review by milana-todorovic — COMMENTED** [review]
Leaving this as a comment since I'm not sure what your exact requirement was, to never have the jmbg in urls (including REST endpoints), or just to not have it in the url for the activation page since that can be seen in the browser.

**Review by milana-todorovic — COMMENTED** [review]
10fe5098a28cde85d73dfaecc9986b915192c31a looks good, although maybe this branch wasn't the best place for it.

File: HealthcareSystem/PatientWebApp/wwwroot/js/activate_account.js (hunk 1 above)
Comment (draganacarapic) [reply HealthcareSystem/PatientWebApp/wwwroot/js/activate_account.js]: Okay, I'll consider moving the decryption to the backend code.

**Comment by draganacarapic** [comment]
[(3) W2.2 As a patient I want to activate my account so that I could use this hospitals application.](https://trello.com/c/P3dRx2Qg/27-3-w22-as-a-patient-i-want-to-activate-my-account-so-that-i-could-use-this-hospitals-application)

File: HealthcareSystem/Backend/Service/Encryption/EncryptionService.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            char[] arr;
+            string result = "";
+            arr = encryptedJmbg.ToCharArray(0, 13);
+
+            foreach (char c in arr)
+            {
+                if (c.ToString() == "A")
+                {
+                    result += "0";
+                }
```
Comment (anciz) [inline HealthcareSystem/Backend/Service/Encryption/EncryptionService.cs:24]: I am wondering why **'c'** and **'arr'**.
(I do not say it is incorrect, but I am confused)

Comment (draganacarapic) [reply HealthcareSystem/Backend/Service/Encryption/EncryptionService.cs:24]: arr is abbreviated from array
c is abbreviated from character

**Review by anciz — DISMISSED** [review]
This is well done!

**Review by milana-todorovic — DISMISSED** [review]
Looks good!

**Review by milana-todorovic — APPROVED** [review]

**Review by Pufke — APPROVED** [review]

**Pull request merged**